uxpin-merge/storybook-material-ui
# Component updates and fixes

**Overview**
original thread: https://uxpin.slack.com/archives/C02GEE0RK5M/p1682618237104099

`storybook-material-ui` is one of our public facing template libraries (`Material UI` in libraries list and part of the [storybook-integration](https://www.uxpin.com/merge/storybook-integration) trial). Because of this, it is important to maintain and update the library when necessary. Currently this library is still on [material ui v4](https://v4.mui.com/getting-started/installation/) but the [git version](https://github.com/uxpin-merge/MUI-Merge/blob/8129ec67f3daede023d4e08b6ae4bca38b0aea77/package.json#LL13C2-L13C2) is on version 5. We should stay consistent as well.

The intention of this PR is to update dependencies, update code, and fix any broken components.

**Fixes**
- [x] Update dependencies (storybook 7, MUI 5, other dependencies)
- [x] Update component code and make any fixes
- [x] Refactor code for modern storybook use
        * storybook controls: https://storybook.js.org/docs/react/essentials/controls
        * MUI + storybook: https://storybook.js.org/recipes/@mui/material

**How to test**
* install `node_modules` with `yarn install`
* Go to UXPin editor and generate a new private storybook library token
* Open terminal in this repo `storybook-material-ui`
* build storybook static `yarn build-storybook`
* push storybook library to UXPin `npx uxpin-storybook-deployer -t TOKEN`
* Go back to editor, open the library and quickly test components on canvas to make sure there are no issues

Side note: 
* Currently this DS is under the `uxpin` organisation. Does it make sense to move this library to `uxpin-merge` organisation and make it public for users?
* The `Datepicker` and `Select` components have popovers that are cut off if the bounding box is too small. Have to resize the bounding box so that the popover will appear correctly in preview.
